pkg/phonebook/user.go

```diff
@@ -46,6 +48,15 @@ func (u *UserAPI) create(r *http.Request) (interface{}, mw.Response) {
 		return nil, mw.BadRequest(err)
 	}
 
+	// Ensure we do not add user that exist in 3bot connect DB with another public key
+	// https://github.com/threefoldtech/home/issues/859
+	if err := u.compareUser(user); err != nil {
+		if errors.Is(err, errWrongPubKey{}) {
```

Comment:
I think you will need to add an implementation of `func (e errWrongPubKey) Is(target error) bool` for this to work, without it, the default behavior is to do an `e == target` for every error in the chain afaik, which will not work due to the missing fields here